cpp/src/arrow/io/file.cc

```diff
@@ -484,6 +491,8 @@ class MemoryMappedFile::MemoryMap : public MutableBuffer {
 
   int64_t size() const { return size_; }
```

Comment:
Is it still useful to expose this? Perhaps `size()` should simply return `map_len_`?

Reply:
@pitrou thanks for the careful review, I'm uncertain on this - is it possible some workload only memory map a file region, but still want to check the whole file size?

Reply:
I don't know. But `GetSize` should return the amount of data that's readable through the file region, not the entire size.

Reply:
I see, will make `GetSize()` return the memory map data length and change the corresponding tests.